examples/util/WebNNRunner.js

```diff
@@ -0,0 +1,383 @@
+   */
+  setSupportedOps = (ops) => {
+    this._supportedOps = ops;
+  };
+
+  /**
+   * This method is to set '_rawModel'.
+   * @param model: An object for model.
+   */
+  _setRawModel = (model) => {
+    this._rawModel = model;
+  };
+
+  /**
+   * This method is to set '_subgraphsSummary'.
+   * @param summary: An array object for summary.
+   */
+  _setSubgraphsSummary = (summary) => {
+    this._subgraphsSummary = summary;
+  };
+
+  /**
+   * This method is to set '_modelRequiredOps'.
+   * @param ops: An array object for required ops of model.
+   */
+  _setModelRequiredOps = (ops) => {
+    this._modelRequiredOps = ops;
+  };
+
+  /**
+   * This method is to set '_deQuantizeParams'.
+   * @param params: An array object for deQuantize params.
+   */
+  _setDeQuantizeParams = (params) => {
+    this._deQuantizeParams = params;
+  };
+
+  /**
+   * This method is to load model file with specified url.
+   * @param url: A string for model file url.
+   */
+  _loadModelFile = async (url) => {
+    let rawModel = null;
+
+    if (url !== undefined) {
+      const arrayBuffer = await this._loadURL(url, this._progressHandler, true);
+      const bytes = new Uint8Array(arrayBuffer);
+      switch (url.split('.').pop()) {
+        case 'tflite':
+          const flatBuffer = new flatbuffers.ByteBuffer(bytes);
+          rawModel = tflite.Model.getRootAsModel(flatBuffer);
+          rawModel._rawFormat = 'TFLITE';
+          printTfLiteModel(rawModel);
+          break;
+        case 'onnx':
+          const err = onnx.ModelProto.verify(bytes);
+          if (err) {
+            throw new Error(`The model file ${url} is invalid, ${err}`);
+          }
+          rawModel = onnx.ModelProto.decode(bytes);
+          rawModel._rawFormat = 'ONNX';
+          printOnnxModel(rawModel);
+          break;
+        case 'bin':
+          const networkFile = url.replace(/bin$/, 'xml');
+          const networkText = await this._loadURL(networkFile);
+          const weightsBuffer = bytes.buffer;
+          rawModel = new OpenVINOModel(networkText, weightsBuffer);
+          rawModel._rawFormat = 'OPENVINO';
+          break;
+        default:
+          throw new Error(`Unrecognized model format, support TFLite | ONNX | OpenVINO model`);
+      }
+    } else {
+      throw new Error(`There's none model file info, please check config info of modelZoo.`);
+    }
+
+    this._setRawModel(rawModel);
+    this._setLoadedFlag(true);
+  };
+
+  loadModel = async (modelInfo) => {
+    if (this._bLoaded && this._currentModelInfo.modelFile === modelInfo.modelFile) {
+      console.log(`${this._currentModelInfo.modelFile} already loaded.`);
+      return;
+    }
+
+    // reset all states
+    this._setLoadedFlag(false);
+    this._setInitializedFlag(false);
+    this._setBackend(null);
+    this._setPrefer(null);
+    this._setModelInfo(modelInfo);
+    this._setModelRequiredOps(new Set());
+    this._setDeQuantizeParams([]);
+    this._setSubgraphsSummary([]);
```

Comment:
those initializes could move to constructor.

Reply:
Since the runner instance was created by single mode, so when we change model, we need do this initializes on loadModel method.

Reply:
That's fine. Please introduce a `_doIntialization` method that sub-classes can override.